scripts/ppr.py

```diff
@@ -0,0 +1,131 @@
+import psycopg2
+import requests
+import json
+import os
+
+def insertPath(cur, relation_id, dhid_from, dhid_to, path):
+    stepList = [f"{step[0]} {step[1]}" for step in path]
+    linestring = "LINESTRING(" + ",".join(stepList) + ")"
+    cur.execute(
+        'INSERT INTO paths (stop_area_relation_id, "from", "to", geom) VALUES (%s, %s, %s, ST_GeomFromText(%s, 4326))',
+        (relation_id, dhid_from, dhid_to, linestring)
+    )
+
+
+def insertPathsElementsRef(cur, edges, path_counter):
+    osm_way_ids = []
+    for edge in edges:
+        osm_way_id = abs(edge["osm_way_id"])
+        # currently also the crossed ways are included (negative way osm id)
+        if osm_way_id != 0 and osm_way_id not in osm_way_ids:
+            osm_way_ids.append(osm_way_id)
+            cur.execute(
+                "INSERT INTO paths_elements_ref (path_id, osm_type, osm_id) VALUES (%s, %s, %s)",
+                (path_counter, 'W', osm_way_id)
+            )
+
+
+def makeRequest(url, payload, stop_places, i, ii):
+    payload["start"]["lat"] = stop_places[i]["lat"]
+    payload["start"]["lng"] = stop_places[i]["lng"]
+    payload["destination"]["lat"] = stop_places[ii]["lat"]
+    payload["destination"]["lng"] = stop_places[ii]["lng"]
+
+    try:
+        response = requests.post(url, json=payload)
+    except Exception as e:
+        exit(e)
+
+    if response.status_code != 200:
+        exit(f'Request failed with status code {response.status_code}: {response.text}')
+
+    return response.json()
+
+
+def insertPGSQL(cur,json_data,stop_places,path_counter,i,ii):
+    # PPR can return multiple possible paths for one connection:
+    for route in json_data["routes"]:
+        path = route["path"]
+        edges = route["edges"]
+        # distance = route["distance"]
+        relation_id = stop_places[ii]["relation_id"]
+                
+        insertPathsElementsRef(cur, edges, path_counter)
+        insertPath(cur, relation_id, stop_places[i]["IFOPT"], stop_places[ii]["IFOPT"], path)
+
```

Comment:
Just some minor coding style suggestion.
Instead of passing the array with indexes it seems more common to me to directly pass the actual values/contents of the array at these indexes. But maybe you have a good reason for not doing so.

Reply:
Thank you for the suggestion, i have changed this section in commit 85c5f97.